get_block_details.py

```diff
@@ -0,0 +1,189 @@
+            block_value = v.get_value()
+            values.append(block_value)
+    return values
+
+
+def set_rc_values_for_block_from_pvs(block_object, pvs):
+    """Search pvs for RC values for given block and return them"""
+    block_name = block_object.get_name()
+    items = pvs.items()
+
+    for k, v in items:
+        test_block_name = k.split(':')[0]
+        if block_name == test_block_name:
+            if "LOW.VAL" == k.split(':')[-1]:
+                low_rc = block_object.set_rc_low(v.get_value())
+            if "HIGH.VAL" == k.split(':')[-1]:
+                high_rc = block_object.set_rc_high(v.get_value())
+            if "INRANGE.VAL" == k.split(':')[-1]:
+                inrange_rc = block_object.set_rc_inrange(v.get_value())
+                return low_rc, high_rc, inrange_rc
+
+def set_rc_values_for_blocks(block_objects, pvs):
+    """Set all RC values for all the given blocks"""
+    for object in block_objects:
+        set_rc_values_for_block_from_pvs(object,pvs)
+
+def add_rc_values(pvs, blocks):
+    """Look for RC values in the supplied list o pvs and assign them to the correct block."""
+
+
+def get_info(url):
+    """
+    Reads block information from a url and populates a list of block objects with it.
+
+    Args:
+        url: The block information source.
+
+    Returns: A converted list of block objects.
+
+    """
+    try:
+        page = requests.get(url)
+    except Exception as e:
+        logging.error("URL not found: " + str(url))
+        raise e
+
+    blocks = dict()
+
+    tree = html.fromstring(page.content)
+
+    titles = tree.xpath("//tr/th/a")
+    titles = [t.text for t in titles]
+
+    status = tree.xpath("//table[2]/tbody/tr/td[1]")
+    status_text = list()
+    for s in status:
+        if s.text is None:
+            # This means there is an extra <font> in the XML to change the colour of the value
+            # This only happens when the PV is "Disconnected"
+            # This assumption speeds up the program significantly
+            status_text.append("Disconnected")
+        else:
+            status_text.append(s.text)
+
+    info = tree.xpath("//table[2]/tbody/tr/td[3]")
+
+    assert(len(titles) == len(status_text))
+    assert(len(titles) == len(info))
+
+    for i in range(len(titles)):
+        try:
+            block_raw = info[i].text
+            if block_raw == "null":
+                    value = "null"
+                    alarm = "null"
+
+            else:
+                value_index = 2
+                alarm_index = 3
+                block_split = block_raw.split(None, 3)
+                if len(block_split) < 4:
+                    # Missing a value
+                    continue
+                else:
+                    value = block_split[value_index]
+                    alarm = block_split[alarm_index]
+
+            name = shorten_title(titles[i])
+            status = status_text[i]
+            blocks[name] = Block(name, status, value, alarm, True)
```

Comment:
This try block is huge. We should consider a new function. `parse_block` or `create_block` or something.